doc/production-hosting.md

```diff
@@ -1,40 +1,231 @@
+
+To bootstrap the configuration process,
+[the script that starts the server][start-shields.sh] sets a single
+environment variable:
+
+```
+NODE_CONFIG_ENV=shields-io-production
+```
+
+With that variable set, the server ([using `config`][config]) reads these
+files:
+
+- [`local-shields-io-production.yml`][local-shields-io-production.yml].
+  This file contains secrets which are checked in with a deploy commit.
+- [`shields-io-production.yml`][shields-io-production.yml]. This file
+  contains non-secrets which are checked in to the main repo.
+- [`default.yml`][default.yml]`. This file contains defaults.
+
+[start-shields.sh]: https://github.com/badges/ServerScript/blob/master/start-shields.sh#L7
+[config]: https://github.com/lorenwest/node-config/wiki/Configuration-Files
+[local-shields-io-production.yml]: ../config/local-shields-io-production.example.yml
+[shields-io-production.yml]: ../config/shields-io-production.yml
+[default.yml]: ../config/default.yml
+
+The project ships with `dotenv`, however there is no `.env` in production.
+
+## Badge CDN
+
+Sitting in front of the three servers is a Cloudflare Free account which
+provides several services:
+
+- Global CDN, caching, and SSL gateway for `img.shields.io`
+- Analytics through the Cloudflare dashboard
+- DNS hosting for `shields.io`
+
+Cloudflare is configured to respect the servers' cache headers.
+
+## Frontend
+
+The frontend is served by [GitHub Pages][] via the [gh-pages branch][gh-pages]. SSL is enforced.
+
+`shields.io` resolves to the GitHub Pages hosts. It is not proxied through
+Cloudflare.
+
+Any maintainer can push to `gh-pages` though in practice this needs to be kept
+in sync with the production deploy.
+
+[github pages]: https://pages.github.com/
+[gh-pages]: https://github.com/badges/shields/tree/gh-pages
+
+## Deployment
+
+To set things up for deployment:
+
+1.  Get your SSH key added to the server.
+2.  Clone a fresh copy of the repository, dedicated for deployment.
+    (Not required, but recommended; and lets you use `npm ci` below.)
+3.  Add remotes:
+
+```sh
+git remote add s0 root@s0.shields-server.com:/home/m/shields.git
+git remote add s1 root@s1.shields-server.com:/home/m/shields.git
+git remote add s2 root@s2.shields-server.com:/home/m/shields.git
+```
+
+`origin` should point to GitHub as usual.
+
+4.  Since the deploy uses `git worktree`, make sure you have git 2.5 or later.
+
+To deploy:
+
+1.  Use `git fetch` to obtain a current copy of
+    `local-shields-io-production.yml` from the server (or obtain the current
+    version of that file some other way). Save it in `config/`.
+2.  Check out the commit you want to deploy.
+3.  Run `npm ci`. **This is super important for the frontend build!**
+4.  Run `make deploy-s0` to make a canary deploy.
+5.  Check the canary deploy:
+    - [Visit the server][s0]. Don't forget that most of the preview badges
+      are static!
+    - Look for errors in [Sentry][].
+    - Keep an eye on the [status page][].
```

Comment:
status page link is not showing for me (and I'm not sure what it should be)

Reply:
Fixed!